rust/agama-cli/src/lib.rs

```diff
@@ -138,13 +153,63 @@ async fn build_manager<'a>() -> anyhow::Result<ManagerClient<'a>> {
     Ok(ManagerClient::new(conn).await?)
 }
 
+#[derive(PartialEq)]
+enum InsecureApi {
+    Secure,      // Remote api is secure
+    Insecure,    // Remote api is insecure - e.g. self-signed certificate
+    Forbidden, // Remote api is insecure and its use is forbidden (e.g. user decided not to use it)
+    Unreachable, // Remote api is unrecheable
+}
+
+/// Returns if insecure connection to remote api server is required and user allowed that
+async fn check_remote_api(api_url: String) -> Result<InsecureApi, ServiceError> {
```

Comment:
I think we should fine a better home for this function and the accompanying `InsecureApi`.